src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +153,1304 @@ static DamageTable D_80B65044 = {
+                    Math_Vec3f_Copy(&this->fires[fireIndex].pos, &randomFirePo->actor.world.pos);
+                    randomFirePo->actor.parent = (Actor*)this;
+                    randomFirePo->actor.update = EnBigpo_UpdateFire;
+                    func_800BC154(globalCtx, &globalCtx->actorCtx, &randomFirePo->actor, ACTORCAT_PROP);
+                    randomFirePo->unk20C = fireIndex;
+                    randomFirePo->actor.flags &= ~0x1; // targetable OFF
+                    // make invisible by size:0
+                    Actor_SetScale(&randomFirePo->actor, 0);
+
+                    if (this->actor.child == NULL) {
+                        this->actor.child = &randomFirePo->actor;
+                    } else {
+                        randomFirePo->actor.child = this->actor.child;
+                        this->actor.child = &randomFirePo->actor;
+                    }
+                    break;
+                } else {
+                    randomIndex--;
+                }
+            }
+            enemyPtr = enemyPtr->next;
+        }
+    }
+
+    // remove unused fires
+    for (enemyPtr = FIRST_ENEMY; enemyPtr != NULL; enemyPtr = enemyPtr->next) {
+        if ((enemyPtr->id == ACTOR_EN_BIGPO) && (enemyPtr->params == ENBIGPO_POSSIBLEFIRE)) {
+            randomFirePo = (EnBigpo*)enemyPtr;
+            randomFirePo->actionFunc = EnBigpo_Die;
+            randomFirePo->actor.update = EnBigpo_UpdateFire;
+        }
+    }
+
+    EnBigpo_ChangeToFireCounting(this);
+}
+
+void EnBigpo_ChangeToFireCounting(EnBigpo* this) {
+    this->actionFunc = EnBigpo_FireCounting;
+}
+
+// count fires already found by dampe, once enough: spawn big poe for fight
+void EnBigpo_FireCounting(EnBigpo* this, GlobalContext* globalCtx) {
+    EnBigpo* firePo;
+    s32 activatedFireCount = 0;
+
+    for (firePo = (EnBigpo*)this->actor.child; firePo; firePo = (EnBigpo*)firePo->actor.child) {
+        if ((firePo->actor.params == ENBIGPO_REVEALEDFIRE) && (firePo->actionFunc == EnBigpo_RevealedFireIdle)) {
+            activatedFireCount++;
+        }
+    }
+
+    if (activatedFireCount == 3) { // all fires found
+        EnBigpo_SetupSpawn(this);
+    }
+}
+
+void EnBigpo_SetupFlameCirclePositions(EnBigpo* this, GlobalContext* globalCtx) {
+    EnBigpo* firePo;
+    Vec3f newPos;
+
+    this->idleTimer = 39;
+    for (firePo = (EnBigpo*)this->actor.child; firePo; firePo = (EnBigpo*)firePo->actor.child) {
+        EnBigpo_SetupFlameCircleCutscene(firePo);
+    }
+
+    // set new starting position for flames?
+    if (this->cutsceneCam != NULL) {
+        newPos.x = (Math_SinS(this->actor.yawTowardsPlayer) * 360.0f) + this->actor.world.pos.x;
+        newPos.y = this->actor.world.pos.y + 150.0f;
+        newPos.z = (Math_CosS(this->actor.yawTowardsPlayer) * 360.0f) + this->actor.world.pos.z;
+        func_8016970C(globalCtx, this->cutsceneCam, &this->actor.focus.pos, &newPos);
+    }
+
+    this->actionFunc = EnBigpo_DoNothing;
+}
```

Comment:
```suggestion
void EnBigpo_SetupFlameCirclePositions(EnBigpo* this, GlobalContext* globalCtx) {
    EnBigpo* firePo;
    Vec3f subCamEye;

    this->idleTimer = 39;
    for (firePo = (EnBigpo*)this->actor.child; firePo; firePo = (EnBigpo*)firePo->actor.child) {
        EnBigpo_SetupFlameCircleCutscene(firePo);
    }

    // Setup sub camera: set 'eye' and 'at' and switch to subcamera
    if (this->cutsceneSubCamId != MAIN_CAM) {
        subCamEye.x = (Math_SinS(this->actor.yawTowardsPlayer) * 360.0f) + this->actor.world.pos.x;
        subCamEye.y = this->actor.world.pos.y + 150.0f;
        subCamEye.z = (Math_CosS(this->actor.yawTowardsPlayer) * 360.0f) + this->actor.world.pos.z;
        func_8016970C(globalCtx, this->cutsceneSubCamId, &this->actor.focus.pos, &subCamEye);
    }

    this->actionFunc = EnBigpo_DoNothing;
}
```
This one I need a little more context, I never fully understood `yawTowardsPlayer` but I suspect this is either setting the camera to look at dampe as he's running away or when the camera is turned backwards to look at link: Either way, this is definitely setting up the location and direction of a subcamera:

https://youtu.be/FuwQVTcXdic?list=PLC198DFAC7C398EAD&t=604